lib/reek/smells/smell_detector.rb

```diff
@@ -38,10 +38,13 @@ def contexts
       end
 
       def run_for(context)
-        return [] unless enabled_for?(context)
         return [] if exception?(context)
 
-        sniff(context)
+        if enabled_for?(context)
+          sniff(context)
+        else
+          check_suppression(context)
```

Comment:
Wouldn't that also raise if the smell was disabled via configuration using `exclude`? That would be great, but it also means that we need to output different error message and that we need to make a difference here.
Can you maybe create some cucumber scenarios so we can hash out the details by looking at the concrete scenarios?